### ISSUE TITLE
IDE plugins include correct set of projects based on dependency substitution

### ISSUE THREAD

**Issue by lptr** [prezi/gradle#1]
https://github.com/prezi/gradle/blob/master/design-docs/dependency-substitution.md#story-ide-plugins-include-correct-set-of-projects-based-on-dependency-substitution

This says we need to make sure IDE plugins include the correct dependencies. Does this mean the Gradle plugins for IDEA and Eclipse, or does this mean the built-in support in IntelliJ as well?

**Comment by bigdaz** [comment]
That refers to the project files generated by Gradle using the 'eclipse' and 'idea' plugins. We might want to liaise with JetBrains to assist them to get it working with the built-in IntelliJ support, but that's out of scope for this story.

**Comment by lptr** [comment]
So what is it that we need to fix or check now?

**Comment by bigdaz** [comment]
This story needs to be fleshed out a bit. I guess we need to do substitutions in both directions, and ensure that the generated IDE projects contain the correctly configured modules, as well as the correct library dependencies for each module.

**Comment by bigdaz** [comment]
Feel free to provide a PR for the design spec to flesh out:
- Desired behaviour
- Implementation plan
- Test coverage plan

(That would be the next step)

**Comment by lptr** [comment]
https://github.com/gradle/gradle/pull/411 handles this. I wonder what needs to be updated about the spec.

**Comment by bigdaz** [comment]
The process we like to use is to start with some analysis and thinking about what would need to be tested. Since you're not doing that, we'll need to do it via post-merge review.

It would save me time if you did the analysis and put it in the spec. Otherwise, I'll spend the time thinking about test cases.

I really don't want to have 5 stories "in progress" at the same time. We need to finish some things before starting more. The IDE integration is an example of something we can delay starting.

**Comment by lptr** [comment]
I've sent an email about working on multiple stories and other issues -- let's discuss it there.

I'll add descriptions of test cases to the spec, and a couple of more tests soon. There's only the tests in the PR, no new functionality, so I guess we can merge it right after that, and forget about this story. Do you need anything else from me to merge it?

**Comment by bigdaz** [comment]
Yeah, thanks for the email. I'll reply today.

Re-reading my comment above doesn't make me feel good about how I'm dealing with this. While I didn't mean to express frustration in my comment, it clearly comes across that way. My apologies.

**Comment by lptr** [comment]
Could you reply on https://github.com/gradle/gradle/pull/411? I'll close this issue now.
